students/ericstreit/session03/str_format_lab.py

```diff
@@ -0,0 +1,61 @@
+#Lesson##
+#XXXXX Exercise ##
+#
+#!/usr/bin/env python3
+#define variables
+
+#Task 1
+mytup = ( 2, 123.4567, 10000, 12345.67)
+print("file{:03} : {:.2f} {:.2e} {:.2e}".format(mytup[0], mytup[1], mytup[2], mytup[3]))
+
+#Task 2
+#same as 1 but use an alternative tpe of format string
+print(f'file{mytup[0]:03} : {mytup[1]:.2f} {mytup[2]:.2e} {mytup[3]:.2e}')
+
+
+#Task 3
+mytup = (1,2,3,4,5)
+#define formatter function
+#does not work properly!
+#swaps the first and last tuples????
+#I practically copied this from your video
+def formatter(n):
+    """string exercise task 3"""
+    l = len(n)
+    print(("the {} numbers are: " + ",".join(["{}"] * l).format(l, *n)))
+
```

Comment:
sometimes it's easier to put things on separate lines :-)

```
mytup = (1, 2, 3, 4, 5)


def formatter(n):
    """string exercise task 3"""
    l = len(n)
    format_str = "the {} numbers are: " + ",".join(["{}"] * l)
    return format_str.format(l, *n)

print(formatter(mytup))
```
You had an end parens in the wrong place -- so it was applying teh format to the second part of your string, not the whole thing...